ruby/lib/minitest/queue.rb

```diff
@@ -50,6 +75,21 @@ def requeue!
     end
   end
 
+  module Flakiness
+    # Make failed flaky tests acts as skips for reporters not aware of the difference.
+    def skipped?
+      super || flaked?
+    end
+
+    def flaked?
+      Flaked === failure
+    end
+
+    def mark_as_flaked!
+      self.failures.unshift(Flaked.new(self.failures.shift))
```

Comment:
Just returning `true` to `skipped?` isn't enough, you also need `failures` to contain something with a particular `result_label` otherwise some parts of the Minitest API will still consider this result to be a failure (it doesn't expect this kind of extension TBH).